### ISSUE TITLE
Failing spreadsheet detection case

### ISSUE THREAD

**Issue by jazzido** [tabulapdf/tabula-java#11]
Kind of a pathological case. Detected spreadsheet:

![gsk_2012_q4 page437-1](https://cloud.githubusercontent.com/assets/27584/6578249/7119304c-c719-11e4-8c20-aae086727168.jpg)

Detected rulings:

![gsk_2012_q4 page437-1](https://cloud.githubusercontent.com/assets/27584/6578287/acc38b2e-c719-11e4-8765-1f831a79b75a.jpg)

**Comment by jeremybmerrill** [comment]
Sorry, can you explain what the error is? The rulings all look right to me...

**Comment by jazzido** [comment]
The outer box contains a spreadsheet; the algorithm returns that instead of the actual lattice.

**Comment by jeremybmerrill** [comment]
Gotcha. Does this issue exist in the Ruby implementation?

**Comment by jazzido** [comment]
Apparently, yes

![screen shot 2015-03-11 at 11 10 43 am](https://cloud.githubusercontent.com/assets/27584/6599282/4e30e978-c7df-11e4-8052-683934fff077.png)

**Comment by jazzido** [comment]
I've just remembered: this issue is the reason why I ended up not merging your PR #8, and why #22 still happens. 

Most tables with ruling lines are bounded by a rectangle. If we treat selections as a set of additional ruling lines, we will have this issue a lot.

**Comment by jeremybmerrill** [comment]
Well, we'll have a different issue if we don't treat selections as additional rulings lines (or come up with some other solution).

**Comment by jazzido** [comment]
Yes, but one that I've seen in at most 4 out of the thousands of tables that I've looked at during the past 2 years ;)

**Comment by jeremybmerrill** [comment]
I'm not sure it's so infrequent -- but I think it's also important for users who select the data they want, but don't include the ruling lines. After all, they want to extract the data, not the lines! But we need the exterior ruling lines to do the extraction correctly. 

I'm not opposed to tweaking this feature, I just think we need to keep something like it. Perhaps expanding the selection by N px, but excluding text within that margin? This wouldn't solve the wacky PDF problem. (I've actually long thought we may want to do something that, to include characters partially-contained within the selection.)
